### PR TITLE
Remove dependency on the `winreg` crate

### PR DESCRIPTION
**Pull request by diceride**
Uses the [windows](https://crates.io/crates/windows) crate to search for the installation date in the registry. The aim is to limit the used Windows API crates. 

References
- Windows: [RegGetValueW function](https://learn.microsoft.com/en-us/windows/win32/api/winreg/nf-winreg-reggetvaluew)

### CODE REVIEW

File: src/action/insttime.rs [hunk 1]
```diff
@@ -240,14 +240,39 @@ fn get_install_time() -> Option<SystemTime> {
...
+            Some(install_date_data.as_mut_ptr() as *mut c_void),
+            Some(&mut install_date_size),
+        )
+    };
+
+    if result == NO_ERROR {
+        let install_date = unsafe { install_date_data.assume_init() };
+
+        if install_date > 0 {
+            return Some(SystemTime::UNIX_EPOCH + Duration::from_secs(install_date as u64))
```
Comment (panhania) [inline src/action/insttime.rs:271]: Prefer not to use `as` casts as they don't shield from potential overflow issues. `u64::from` does not have these issues.

**Review by panhania — CHANGES_REQUESTED** [review]

**Comment by diceride** [comment]
- Replaced the windows crate with the lightweight windows-sys crate.
- Used `u64::from` to convert from u32 losslessly.

**Review by panhania — APPROVED** [review]

**Comment by panhania** [comment]
@diceride Looks good now but has merge conflicts. Could you look into resolving them?

**Comment by diceride** [comment]
Bim bam bum late night merge errors and broken Windows vm's are much fun 🐼

**Pull request merged**